Revert to using isolated PID namespaces in Docker

verb opened the pull request:
**What this PR does / why we need it**: Reverts to the previous docker default of using isolated PID namespaces for containers in a pod. There exist container images that expect always to be PID 1 which we want to support unmodified in 1.8.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #48937

**Special notes for your reviewer**:

**Release note**:
<!--  Steps to write your release note:
1. Use the release-note-* labels to set the release note state (if you have access)
2. Enter your extended release note in the below block; leaving it blank means using the PR title as the release note. If no release note is required, just write `NONE`.
-->
```release-note
Sharing a PID namespace between containers in a pod is disabled by default in 1.8. To enable for a node, use the --docker-disable-shared-pid=false kubelet flag. Note that PID namespace sharing requires docker >= 1.13.1.
```

verb committed:
Revert to using isolated PID namespaces in Docker

A shared PID namespace were enabled by default in the 1.7 when running
with a supported Docker runtime, but a Docker version that supports
a shared namespace was not qualified for use.

Release 1.8 will qualify a docker version supporting shared PID, but we
don't want to cause disruption for container images which expect always
to have PID 1.

verb commented:
looks like maybe the e2e node test doesn't set the flag. Will look into it tomorrow.

derekwaynecarr commented:
poke when tests pass, thanks for making the update.

dims commented:
@verb do we want to specify the docker versions needed for the shared pid flag to work in the release note?

verb commented:
@dims oh yes, great catch. Updated the release note.

verb commented:
@yguo0905 can you offer any advice about how to best to update the node e2e test for this? It looks like the kubelet is only run one time for the test, so I can't test with both the flag enabled and disabled.

Should I test the default behavior or test the feature even though it's disabled by default?

I assume the former but I want to make sure I'm not missing something.

yguo0905 commented:
I think we want to test this behavior because user can enable it if they want. We can set the flag at https://github.com/kubernetes/kubernetes/blob/55cb667b9ec975664609e736f49713ee24419b81/test/e2e_node/jenkins/jenkins-pull.properties#L7

verb commented:
/retest

verb commented:
/retest

yguo0905 commented on test/e2e_node/services/kubelet.go:
```diff
@@ -155,6 +155,7 @@ func (e *E2EServices) startKubelet() (*server, error) {
 		"--serialize-image-pulls", "false",
 		"--pod-manifest-path", manifestPath,
 		"--file-check-frequency", "10s", // Check file frequently so tests won't wait too long
+		"--docker-disable-shared-pid=false",
```
If we set it here we don't need the change in `test/e2e_node/jenkins/jenkins-pull.properties`?

Actually, if we don't set it here there are other `properties` files in `test/e2e_node/jenkins/` need to be changed.

verb replied:
Ok, thanks. Based on this advice (and because it made the e2e node test pass) I've removed the jenkins config and set the flag here.

verb committed:
Explicitly enable docker shared-pid for e2e_node

This also renames isSharedPIDNamespaceEnabled() to
isSharedPIDNamespaceSupported() to be more accurate.

verb commented:
/retest

>

verb commented:
/retest

verb commented:
/retest

verb commented:
This PR is ready for review. According to sig-testing, pull-kubernetes-e2e-kops-aws is broken and has been disabled for the merge queue.

This PR should be added to the 1.8 milestone to resolve critical-urgent issue #48937.

dims commented:
@kubernetes/sig-release-members Can you please apply milestone v1.8?

derekwaynecarr commented:
/lgtm
/approve

verb commented:
/retest

jdumars commented:
@verb could you please assign a SIG to this?

verb commented:
/sig node
